odl/trafos/wavelet.py

```diff
@@ -606,23 +912,47 @@ def __init__(self, dom, nscales, wbasis, mode):
             raise ValueError('Cannot use more than {} scaling levels, '
                              'got {}.'.format(max_level, self.nscales))
 
-        self.size_list = coeff_size_list(dom.shape, self.nscales,
-                                         self.wbasis, self.mode)
-
-        ran_size = np.prod(self.size_list[0])
-
-        if dom.ndim == 1:
-            ran_size += sum(np.prod(shape) for shape in
-                            self.size_list[1:-1])
-        elif dom.ndim == 2:
-            ran_size += sum(3 * np.prod(shape) for shape in
-                            self.size_list[1:-1])
-        elif dom.ndim == 3:
-            ran_size += sum(7 * np.prod(shape) for shape in
-                            self.size_list[1:-1])
+        self._axes = axes
```

Comment:
The argument should be sanitized, i.e. something like
```python
self._axes = tuple(int(ax) for ax  in axes)
```
This ensures that the attribute can be *guaranteed* to be of a certain type rather than user input dependent.
On top of that, `axes` is an important attribute of the transform, make it public (add a property).